src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/InitializationService.java

```diff
@@ -0,0 +1,315 @@
+     * @throws UnhandledWarningsException
+     */
+    private void checkForWarnings(ClientId ownerClientId, String securityServerCode)
+            throws UnhandledWarningsException {
+        String ownerMemberName = globalConfFacade.getMemberName(ownerClientId);
+        SecurityServerId serverId = SecurityServerId.create(ownerClientId, securityServerCode);
+        List<WarningDeviation> warnings = new ArrayList<>();
+        if (StringUtils.isEmpty(ownerMemberName)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_UNREGISTERED_MEMBER,
+                    ownerClientId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (globalConfFacade.existsSecurityServer(serverId)) {
+            WarningDeviation memberWarning = new WarningDeviation(WARNING_INIT_SERVER_ID_EXISTS,
+                    serverId.toShortString());
+            warnings.add(memberWarning);
+        }
+        if (!warnings.isEmpty()) {
+            throw new UnhandledWarningsException(warnings);
+        }
+    }
+
+    /**
+     * Helper to create an initial client
+     * @param clientId
+     * @return
+     */
+    private ClientType getInitialClient(ClientId clientId) {
+        ClientType localClient = clientService.getLocalClient(clientId);
+        if (localClient == null) {
+            localClient = new ClientType();
+            localClient.setIdentifier(clientId);
+            localClient.setClientStatus(ClientType.STATUS_SAVED);
+            localClient.setIsAuthentication(IsAuthentication.SSLAUTH.name());
+        }
+        return localClient;
+    }
+
+    /**
+     * Verify that the initialization process can proceed and that the security server has not already been
+     * initialized. This means verifying that an anchor has been imported,
+     * server conf does not exist and a software token has not yet been initialized. This method could also be
+     * changed into throwing an UnhandledWarningsException with warning metadata instead of the current
+     * InitializationException if partial init needs to be supported in the future
+     * @throws InitializationException if server conf exists OR software token is already initialized
+     */
+    private void verifyInitializationPrerequisites() throws InitializationException {
+        boolean isServerConfInitialized = serverConfService.isServerConfInitialized();
+        boolean isSoftwareTokenInitialized = tokenService.isSoftwareTokenInitialized();
+        List<String> metadata = new ArrayList<>();
+        if (isServerConfInitialized) {
+            metadata.add(METADATA_SERVERCONF_EXISTS);
+        }
+        if (isSoftwareTokenInitialized) {
+            metadata.add(METADATA_SOFTWARE_TOKEN_INITIALIZED);
+        }
+        if (!metadata.isEmpty()) {
+            throw new InitializationException("Error initializing security server", metadata);
+        }
+    }
+
+    /**
+     * If something goes south with the initialization
```

Comment:
nitpick: maybe instead of "going south" something more neutral, since it is a bit conversational and codebase is sort of official and open source (this was an extreme nitpick, sorry)

Reply:
Did not even think about this. Fixed.

Checking from Wiktionary, the idiom seems to have quite a few different meanings - some of which have darker or otherwise inappropriate tone to them.

Reply:
:)